### PR TITLE
Issue #11446: Update CheckerTest method testUnmappableCharacters

### PR DESCRIPTION
**Pull request by chrccl**
Issue #11446: Update CheckerTest method testUnmappableCharacters

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java [hunk 1]
```diff
@@ -1659,20 +1666,13 @@ public void testCachedFile() throws Exception {
...
         final String[] expected = {
-            "4: " + getCheckMessage(LineLengthCheck.class, MSG_KEY, 75, 238),
+            "9: " + getCheckMessage(LineLengthCheck.class, MSG_KEY, 80, 100),
         };
 
-        final DefaultConfiguration checkConfig = createModuleConfig(LineLengthCheck.class);
-        checkConfig.addProperty("max", "75");
-
-        final DefaultConfiguration checkerConfig = createRootConfig(checkConfig);
-        checkerConfig.addProperty("charset", "IBM1098");
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:1673]: Please use xml formatted config in Input file, and set this property to Checker.

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java [hunk 2]
```diff
@@ -429,13 +428,21 @@ public void testSetters() {
         checker.setFileExtensions(".java", "xml");
 
         try {
-            checker.setCharset("UNKNOWN-CHARSET");
+            checker.setCharset("invalid_charset_name");
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:431]: Why this change is required?

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java (hunk 1 above)
Comment (chrccl) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:1673]: Good evening @romani , I tried to write the properties in xml in Input file but it gives me "Checkstyle Config comment not specified properly". This is my config comment:

`/*xml
<module name="Checker">
  <property name="charset" value="IBM1098"/>
  <module name="LineLength">
    <property name="max" value="75"/>
  </module>
</module>

*/`

Could you help me please? Did I write the xml wrong?

Comment (chrccl) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:1673]: GoodMorning @romani, 
I debugged it:

[
Screenshot 2024-02-17 115924
](url)

The problem was that the "getFullyQualifiedClassName" method was unable to return the "FullyQualifiedClassName" for "LineLength." So, since the method expects it, I passed the FullyQualifiedName directly in the XML configuration. This change makes the test pass. If passing the FullyQualifiedName directly doesn't seem elegant as a solution, I can add another entry to the Map "moduleMappings" so that I can obtain the FullyQualifiedName for "LineLength" within the "getFullyQualifiedClassName" method instead of passing it directly.

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java (hunk 2 above)
Comment (chrccl) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java]: In fact, for the purposes of the issue, it is not required... "UNKNOWN-CHARSET" could also work fine.

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java (hunk 1 above)
Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:1673]: nice!!!

**Review by romani — APPROVED** [review]
ok to merge if CI pass

**Review by romani — CHANGES_REQUESTED** [review]
please investigate test coverage reduction
https://github.com/checkstyle/checkstyle/actions/runs/7941288117/job/21683472819?pr=14492

it is blocker to merge

**Comment by chrccl** [comment]
@romani ping

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java [hunk 3]
```diff
@@ -1659,20 +1658,13 @@ public void testCachedFile() throws Exception {
...
         };
 
-        final DefaultConfiguration checkConfig = createModuleConfig(LineLengthCheck.class);
-        checkConfig.addProperty("max", "75");
-
-        final DefaultConfiguration checkerConfig = createRootConfig(checkConfig);
-        checkerConfig.addProperty("charset", "IBM1098");
-
-        verify(checkerConfig, getPath("InputCheckerTestCharset.java"), expected);
+        verifyWithInlineConfigParser(getPath("InputCheckerTestCharset.java"), expected);
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:1667]: please reuase `verifyWithInlineXmlConfig` method, it fully supports xml configs

**Review by romani — CHANGES_REQUESTED** [review]
item:

**Comment by romani** [comment]
@chrccl , ping

**Comment by nrmancuso** [comment]
We have lost contact with author, anyone is welcome to reuse changes here to close the issue.
